src/EcomDestinationWriter.cs

```diff
@@ -1691,9 +1714,77 @@ private void WritePrices(Dictionary<string, object> row, Dictionary<string, Colu
+            if (string.IsNullOrWhiteSpace(priceCurrencyValue))
+            {
+                row[priceCurrencyColumn.SourceColumn.Name] = Ecommerce.Services.Currencies.GetDefaultCurrency().Code;
+            }
+        }
+
+        if (columnMappings.TryGetValue("PriceUserId", out var priceAccessUserColumn))
+        {
+            var userIdLookupValue = GetMergedValue(priceAccessUserColumn, row);
+            var userIDs = ExistingUsers.Select("AccessUserID='" + userIdLookupValue + "'").Select(r => r["AccessUserID"].ToString()).ToList();
+            if (userIDs.Count == 0)
+            {
+                userIDs = ExistingUsers.Select("AccessUserExternalID='" + userIdLookupValue + "'").Select(r => r["AccessUserID"].ToString()).ToList();
+            }
+
+            if (userIDs.Count == 1)
+            {
+                dataRow["PriceUserId"] = userIDs[0];
+                row["PriceUserId"] = userIDs[0];
+            }
+        }
+        if (columnMappings.TryGetValue("PriceUserGroupId", out var priceAccessUserGroupColumn))
+        {
+            var userIDs = ExistingUsers.Select("AccessUserExternalID='" + GetMergedValue(priceAccessUserGroupColumn, row) + "'").Select(r => r["AccessUserID"].ToString()).ToList();
+            if (userIDs.Count == 1)
+            {
+                dataRow["PriceUserGroupId"] = userIDs[0];
+                row["PriceUserGroupId"] = userIDs[0];
+            }
+        }
+    }
+
+    private void WriteDiscounts(Dictionary<string, object> row, Dictionary<string, ColumnMapping> columnMappings, DataRow dataRow)
+    {
+        if (columnMappings.TryGetValue("DiscountAccessUser", out var discountAccessUserColumn))
+        {
+            var userIDs = ExistingUsers.Select("AccessUserExternalID='" + GetMergedValue(discountAccessUserColumn, row) + "'").Select(r => r["AccessUserID"].ToString()).ToList();
+            if (userIDs.Count == 1)
+            {
+                dataRow["DiscountAccessUserId"] = userIDs[0];
+                row["DiscountAccessUserId"] = userIDs[0];
+            }
+        }
+        if (columnMappings.TryGetValue("DiscountAccessUserGroup", out var discountAccessUserGroupColumn))
+        {
+            var userIDs = ExistingUsers.Select("AccessUserExternalID='" + GetMergedValue(discountAccessUserGroupColumn, row) + "'").Select(r => r["AccessUserID"].ToString()).ToList();
+            if (userIDs.Count == 1)
+            {
+                dataRow["DiscountAccessUserGroupId"] = userIDs[0];
+                row["DiscountAccessUserGroupId"] = userIDs[0];
+            }
+        }
+
+        if (!columnMappings.TryGetValue("DiscountCurrencyCode", out var discountCurrencyColumn))
+        {
+            dataRow["DiscountCurrencyCode"] = Ecommerce.Services.Currencies.GetDefaultCurrency().Code;
+        }
+        else
+        {
+            var discountCurrencyValue = (string)row[discountCurrencyColumn.SourceColumn.Name];
```

Comment:
same as above, what if source column is null?

Reply:
sure, yeah we never know what will happend with dw10, so will add null-checks :)